nagios/bin/pmp-check-mysql-deleted-files

```diff
@@ -99,7 +101,7 @@ main() {
 # command name to match.
 # ########################################################################
 _pidof() {
-   if ! pidof "${1}" 2>/dev/null; then
+   if ! pidof -s "${1}" 2>/dev/null; then
```

Comment:
Is it?  Running pidof without the flag returns something like this:

```
$ pidof bash
31422 28773 24239 23586 20371 20123 17350 11149 1561
```

This whole string then gets feed into `head -n1`, which does nothing because it's a space-delimited string (not newline-delimited, like the ack code).  Therefore, `$PROC_ID` is declared as this large string.

This then gets passed to `_lsof`, which would return something like this:

```
$ lsof -p `pidof bash`
lsof: status error on 28773: No such file or directory
lsof: status error on 24239: No such file or directory
lsof: status error on 23586: No such file or directory
lsof: status error on 20371: No such file or directory
lsof: status error on 20123: No such file or directory
lsof: status error on 17350: No such file or directory
lsof: status error on 11149: No such file or directory
lsof: status error on 1561: No such file or directory
lsof 4.82
 latest revision: ftp://lsof.itap.purdue.edu/pub/tools/unix/lsof/
 latest FAQ: ftp://lsof.itap.purdue.edu/pub/tools/unix/lsof/FAQ
 latest man page: ftp://lsof.itap.purdue.edu/pub/tools/unix/lsof/lsof_man
 usage: [-?abhlnNoOPRtUvVX] [+|-c c] [+|-d s] [+D D] [+|-f[gG]] [+|-e s]
 [-F [f]] [-g [s]] [-i [i]] [+|-L [l]] [+m [m]] [+|-M] [-o [o]] [-p s]
[+|-r [t]] [-s [p:s]] [-S [t]] [-T [t]] [-u s] [+|-w] [-x [fl]] [--] [names]
Use the ``-h'' option to get more help information.
```

Ergo, not including `-s` is a bug, because the lsof output is now useless.  Worse, it's a silent bug because of the `2> /dev/null` piece.

Reply:
Apologies, I was thinking that it behaved like `pgrep` .. so, what about the following options:

- instead of `-s` you apply ` | tr ' ' '\n'` to support the `head -n1`, or
- support multiple PIDs with something like ` | tr ' ' ,`

Reply:
Sounds reasonable.  The comma-delimited format would work for `lsof`, but I'll need to figure out some solutions for the other alternatives.

The `ls` command could have `/proc/{123,456}/fd`.  Comma-delimited brace expansion should be fine for old-school `sh`.  I might have to use a single proc for `fstat`.  Don't have a BSD box to test against, either.